tests/python/relay/test_op_level2.py

```diff
@@ -0,0 +1,40 @@
+import tvm
+from tvm import relay
+
+
+def test_conv2d_infer_type():
+    # symbolic in batch dimension
+    ib = relay.ir_builder.IRBuilder()
+    n, c, h, w = tvm.var("n"), 10, 224, 224
+    x = ib.param("x", relay.ty.TensorType((n, c, h, w), "float32"))
+    w = ib.param("w", relay.ty.IncompleteType())
+
+    with ib.function(x, w) as func:
+        ib.ret(relay.nn.conv2d(x.var, w.var,
+                               kernel_size=(3, 3),
+                               padding=(1, 1),
+                               channels=2))
+    ib.ret(func)
+    func = relay.ir_pass.infer_type(ib.env, func.to_func())
+    ftype = func.checked_type()
+    assert ftype.ret_type == relay.ty.TensorType(
```

Comment:
@tqchen 
I tried some hands on with this test case and observed this comparison hold good only for the dtype not for shape. 

Am I missing some thing here?

Reply:
Can you elaborate? The testcase should works for shape as well

Reply:
Great catch, this is actually due to a problem in alpha_eq comparator, I have fixed this in recent commit